crates/ide/src/references/rename.rs

```diff
@@ -20,17 +20,51 @@ use crate::{
     SourceChange, SourceFileEdit, TextRange, TextSize,
 };
 
+#[derive(Debug)]
+pub struct RenameError(pub(crate) String);
+
+impl RenameError {
+    fn new(message: &str) -> RenameError {
```

Comment:
As per https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#premature-pessimization, this should be `message: String`. But given that would be just `RenameError()`, I think we should just remove the `new` function.